src/videojs-contrib-hls.js

```diff
@@ -328,6 +330,10 @@ class HlsHandler extends Component {
       this.selectPlaylist ?
         this.selectPlaylist.bind(this) : Hls.STANDARD_PLAYLIST_SELECTOR.bind(this);
 
+    this.masterPlaylistController_.initialSelectPlaylist =
+      this.selectPlaylist ?
```

Comment:
After a source change, this will set the `initialSelectPlaylist` to be the standard selector and not the initial one. You should add another properties to the source handler for `selectInitialPlaylist` and then this becomes `this.selectInitialPlaylist ? this.selectInitialPlaylist.bind(this) : Hls.INITIAL_PLAYLIST_SELECTOR.bind(this)`.

side note, by my snippet you'll see i personally prefer `selectInitialPlaylist` over `initialSelectPlaylist`, but either is descriptive and fine, so feel free to change or keep the naming. @gesinger probably has his own preference